### PR TITLE
Edit form

### PR DESCRIPTION
**Pull request by neilbulloch**

### CODE REVIEW

File: edit.php [hunk 1]
```diff
@@ -2,30 +2,30 @@
...
-
-$query = $db->prepare($sql);
-$query->bindParam(':title', $title);
-$query->bindParam(':imgUrl', $imgUrl);
-$query->bindParam(':siteUrl', $siteUrl);
-$query->bindParam(':id', $id);
-$query->execute();
+    $sql = "UPDATE `projects` SET `title`= :title, `img_url`= :imgUrl, `site_url`= :siteUrl WHERE `id`= :id ;";
 
+    $query = $db->prepare($sql);
```
Comment (alicemcneil) [inline edit.php]: could add PDO:: to define input data type for more security?

**Review by alicemcneil — COMMENTED** [review]
looks good to me! 
there is a file called editForm.php in your dir that doesn't do anything - could be deleted?

File: edit.php (hunk 1 above)
Comment (neilbulloch) [reply edit.php]: Sam said it was ok to take it out for this one

File: admin.php [hunk 2]
```diff
@@ -2,7 +2,9 @@
 $db = new PDO("mysql:host=192.168.20.20;dbname=Portfolio", 'root', '');
 $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
 
-$sql = 'SELECT `id`,`title`, `img_url`, `date_added` FROM `projects`;';
+
```
Comment (mporam) [inline admin.php]: why extra lines?

File: admin.php [hunk 3]
```diff
@@ -40,8 +42,8 @@
     echo '<tr><td>' . $project['title'] . '</td>';
     echo '<td>' . $project['img_url'] . '</td>';
     echo '<td>' . $project['date_added'] . '</td>';
-    echo '<td><a href="edit.php?' . $project['id'] . '" class="admin-button edit">Edit</a></td>';
-    echo '<td><a href="#" class="admin-button delete">Delete</a></td></tr>';
+    echo '<td><a href="edit.php?id=' . $project['id'] . '" class="admin-button edit">Edit</a></td>';
+    echo '<td><a href="delete.php?id=' . $project['id'] . '" class="admin-button delete">Delete</a></td></tr>';
```
Comment (mporam) [inline admin.php]: not related to editing

File: edit.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+$db = new PDO("mysql:host=192.168.20.20;dbname=Portfolio", 'root', '');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+if (!empty($_GET['id'])) {
+    $sql = 'SELECT `id`, `title`, `img_url`, `site_url` FROM `projects` WHERE `id`= ?;';
+    $query = $db->prepare($sql);
+    $query->execute([$_GET['id']]);
+    $result = $query->fetch();
+} else {
+    echo 'ERROR: $_GET[id] is empty';
```
Comment (mporam) [inline edit.php]: generally we try to make error messages not techy, so we dont give away information to hackers

File: edit.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+if(!isset($_POST['cancelBtn'])) {
```
Comment (mporam) [inline edit.php]: can we split this code into a seperate file? rather than making the edit form submit to itself

File: edit.php [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    <title>Edit project entry</title>
+</head>
+<body>
+<div class="logo">
+    <img src="Images/logo_transparent.png" alt="logo">
+</div>
+<h1>Edit Project</h1>
+
+<div class="addNewForm">
+    <form method = "POST" enctype="multipart/form-data">
```
Comment (mporam) [inline edit.php]: add an action and point the form at a different file please

File: edit.php [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        <?php
+            echo '<input type="hidden" name="id" value="'. $result['id'] . '"</label>';
+            echo '<label>Project Title: </label>';
+            echo '<input type="text" value="' . $result['title'] . '" name="title">';
+            echo '<label>Project URL: </label>';
+            echo '<input type="text" value="' . $result['site_url'] . '" name="site">';
+            echo '<label>Image URL: </label>';
+            echo '<input type="text" value="' . $result['img_url'] . '" name="image">';
```
Comment (mporam) [inline edit.php]: instead of showing the image path as a text url, you could show the image as an image tag, nicer UX

**Review by mporam — CHANGES_REQUESTED** [review]

**Pull request merged**